src/client/finCsrfFetch.ts

```diff
@@ -98,7 +99,7 @@ async function getAttestKey(
 	} catch (err) {
 		// If the promise fails then clear the entry from the cache so it can be requested again in future.
 		originAttestKeyPromises.delete(host)
-		throw err
+		throw new AuthenticationError(err as Error)
```

Comment:
I agree. this should be an HTTP Error here. Even though it may ultimately result in an AuthenticationError, Authentication Error should be thrown from the authenticator if it detects an auth error.